v-next/hardhat/templates/node-test-runner-viem/test/Lock.ts

```diff
@@ -1,11 +1,11 @@
-import { describe, it, before } from "node:test";
+import { describe, it } from "node:test";
 import assert from "node:assert/strict";
-import hre from "@ignored/hardhat-vnext";
+import { network } from "@ignored/hardhat-vnext";
 import { getAddress, parseGwei } from "viem";
-import { NetworkConnection } from "@ignored/hardhat-vnext/types/network";
 
-describe("Lock", function () {
-  let networkConnection: NetworkConnection<"l1">;
+describe("Lock", async function () {
+  const { viem, networkHelpers } = await network.connect();
```

Comment:
What do you think about this async operation in the describe? @kanej @galargh 

It makes the tests way simpler.

Reply:
I think it cleans up the code. But I do find it surprising - I expect resource initialization code to be in the before/beforeEach.

I was initially concerned that this code might get run even if the describe is skipped, but I can confirm that isn't the case.

Would we use a similar pattern in the `mocha` template?

Reply:
`mocha` intentionally doesn't support that. So we can't do it there.

Maybe we can initialize the network connection on the first provider call? I think the code style improvement is massive.

Reply:
In reality, given that this is ESM, these calls can be in the top-level of the file.